### PR TITLE
[hotfix] Fix find last process instance in dependent task is not correct

### PR DESCRIPTION
**Pull request by ruanwenjun**
## Verify this pull request

This pull request is code cleanup without any test coverage.

*(or)*

This pull request is already covered by existing tests, such as *(please describe tests)*.

(or)

This change added tests and can be verified as follows:

(or)

If your pull request contain incompatible change, you should also add it to `docs/docs/en/guide/upgrede/incompatible.md`

### CODE REVIEW

**Comment by caishunfeng** [comment]
PTAL @lenboo @EricJoy2048

File: dolphinscheduler-master/src/main/java/org/apache/dolphinscheduler/server/master/utils/DependentExecute.java [hunk 1]
```diff
@@ -187,12 +187,6 @@ private DependResult getDependTaskResult(long taskCode, ProcessInstance processI
      */
     private ProcessInstance findLastProcessInterval(Long definitionCode, DateInterval dateInterval) {
 
-        ProcessInstance runningProcess = processService.findLastRunningProcess(definitionCode,
-                dateInterval.getStartTime(), dateInterval.getEndTime());
-        if (runningProcess != null) {
-            return runningProcess;
-        }
-
```
Comment (brave-lee) [inline dolphinscheduler-master/src/main/java/org/apache/dolphinscheduler/server/master/utils/DependentExecute.java:195]: I want to know why to remove this code

Comment (ruanwenjun) [reply dolphinscheduler-master/src/main/java/org/apache/dolphinscheduler/server/master/utils/DependentExecute.java:195]: AFAIK, this method is just want to find the latest `processInstance` by `endTime` or `scheduleTime`.
```
find the last one process instance that :
     * 1. manual run and finish between the interval
     * 2. schedule run and schedule time between the interval
     private ProcessInstance findLastProcessInterval(Long definitionCode, DateInterval dateInterval)
```
It doesn't care about the before processInstance.

For example: 
If I have a process instance manual executed 5 times today, and I use this method to query today process instance.
```
11:00 finished
10:00 finished
9:00 running
8:00 finished
7:00 finished
```
This method should return the latest processInstance(11:00), not 9:00.

Comment (brave-lee) [reply dolphinscheduler-master/src/main/java/org/apache/dolphinscheduler/server/master/utils/DependentExecute.java:195]: Yep, you're right. There are indeed this bug.

**Review by brave-lee — APPROVED** [review]
+1

**Comment by brave-lee** [comment]
Please pick up to 2.0.7

**Comment by 981433814** [comment]
Does this change also apply to version 2.x? I see that version 2.x also has this problem

**Pull request merged**